tests/protocol/src/bin.c

```diff
@@ -985,6 +990,100 @@ ZTEST(thingset_bin, test_import_data)
+        "10 19 03 E8 "             /* t_s */
+        "19 02 01 F5 "             /* Types/wBool */
+        "19 06 00 02 "             /* Records: 2 */
+        "19 07 01 01 "             /* Nested/rBeginning */
+        "19 07 08 FA 40 0C CC CD"; /* Nested/Obj2/rItem2_V */
+
+    struct thingset_context ts_local;
+    thingset_init(&ts_local, data_objects, data_objects_size);
+
+    int data_len = hex2bin_spaced(rpt_exp_hex, data, sizeof(data));
+
+    /* save old values */
+    uint32_t timestamp_old = timestamp;
+    bool b_old = b;
+    int32_t nested_beginning_old = nested_beginning;
+    float nested_obj2_item2_old = nested_obj2_item2;
+
+    /* set values to zero/false */
+    timestamp = 0;
+    b = false;
+    nested_obj2_item2 = 0.0;
+    nested_beginning = 0;
+
+    int status = thingset_import_report(&ts_local, data, data_len, THINGSET_WRITE_MASK,
+                                        THINGSET_BIN_IDS_VALUES, 0x800);
+    zassert_equal(status, 0);
+    zassert_equal(timestamp, timestamp_old);
+    zassert_equal(b, b_old);
+    zassert_equal(nested_beginning, nested_beginning_old);
+    zassert_equal(nested_obj2_item2, nested_obj2_item2_old);
+
+    /* set to originial values */
+    timestamp = timestamp_old;
+    b = b_old;
+    nested_beginning = nested_beginning_old;
+    nested_obj2_item2 = nested_obj2_item2_old;
+}
+
+ZTEST(thingset_bin, test_import_report_error)
+{
+
+    uint8_t data[THINGSET_TEST_BUF_SIZE];
+
+    const char rpt_exp_hex[] =
+        "1F 19 08 00 A5 "
+        "10 19 03 E8 "             /* t_s */
+        "19 02 01 F5 "             /* Types/wBool */
+        "19 06 00 02 "             /* Records: 2 */
+        "19 07 01 01 "             /* Nested/rBeginning */
+        "19 07 08 FA 40 0C CC CD"; /* Nested/Obj2/rItem2_V */
+
+    struct thingset_context ts_local;
+    thingset_init(&ts_local, data_objects, data_objects_size);
+
+    int data_len = hex2bin_spaced(rpt_exp_hex, data, sizeof(data));
+
+    /* save old values */
+    uint32_t timestamp_old = timestamp;
+    bool b_old = b;
+    int32_t nested_beginning_old = nested_beginning;
+    float nested_obj2_item2_old = nested_obj2_item2;
+
+    /* set values to zero/false */
+    timestamp = 0;
+    b = false;
+    nested_obj2_item2 = 0.0;
+    nested_beginning = 0;
+
+    /* no report */
+    data[0] = 0x00;
+    int status = thingset_import_report(&ts_local, data, data_len, THINGSET_WRITE_MASK,
+                                        THINGSET_BIN_IDS_VALUES, 0x800);
+    zassert_equal(status, -THINGSET_ERR_UNSUPPORTED_FORMAT);
+
+    /* wrong subset */
+    data[0] = 0x1f;
+    status = thingset_import_report(&ts_local, data, data_len, THINGSET_WRITE_MASK,
+                                    THINGSET_BIN_IDS_VALUES, 0x0);
+    zassert_equal(status, -THINGSET_ERR_NOT_FOUND);
+
+    /* set to originial values */
+    timestamp = timestamp_old;
+    b = b_old;
+    nested_beginning = nested_beginning_old;
+    nested_obj2_item2 = nested_obj2_item2_old;
```

Comment:
This test should not change the values. Should we maybe `assert` that the new values are equal to the old values instead?